code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OfferExtension.kt

```diff
@@ -0,0 +1,28 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.optimize
+
+internal object OfferExtension {
+    @JvmStatic
+    fun List<Offer>?.toUniquePropositionsList(): List<OptimizeProposition>? {
```

Comment:
The logic here in conjunction with `OptimizeUtils.generateInteractionXdm` doesn't achieve our goal. You're creating a list of proposition reference for the list of offers passed in. Then in `OptimizeUtils.generateInteractionXdm` you're adding all the offer ids in the proposition to the display events, rather only the ids of List<Offer> that was passed to the API.
https://github.com/siddique-adobe/aepsdk-optimize-android/pull/3/files#diff-b90c589ff2ce5c066bb12f60a6b2ce62e77b4fede59cbe980f944bd99f45680dR288

Reply:
Please review this again, I've fixed the missing logic here @spoorthipujariadobe